lib/libpng_sb/libpng_sb.c

```diff
@@ -95,8 +95,8 @@ libpng_sb_userfn_handler(struct cheri_object system_object __unused,
     __capability void *c5 __unused, __capability void *c6 __unused,
     __capability void *c7 __unused)
 {
-	struct sb_png_struct *psp = (struct sb_png_struct *)void_cpsp;
-
+	struct sb_png_struct *psp = (struct sb_png_struct *)cheri_cap_to_ptr(void_cpsp);
+	void* c4_ptr = cheri_cap_to_ptr(c4); /* XXXAR: check cap bounds? */
```

Comment:
Maybe `cheri_cap_to_ptr()` should take a minimum size and return `NULL` when space isn't available?

Reply:
I consindered that and think I will do it. If there is anywhere where we don't know/don't care I can add a `cheri_cap_to_ptr_unsafe()` version.

Reply:
You can always ask for at least 0.